core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -44,6 +44,20 @@ sealed trait Chunk[+A] { self =>
    */
   def collect[B](p: PartialFunction[A, B]): Chunk[B] = self.materialize.collect(p)
 
+  /**
+   * Transforms all elements of the chunk for as long as the specified partial function is defined.
+   */
+  def collectWhile[B](p: PartialFunction[A, B]): Chunk[B] = {
+    val len = self.length
+
+    var i = 0
+    while (i < len && p.isDefinedAt(self(i))) {
```

Comment:
This needs to be formulated with `pf.applyOrElse` to avoid double evaluation of the PartialFunction

Reply:
I'm aware, same thing happens in `Arr#collect`. Should I fix it too?

Reply:
That'd be good!

Reply:
Done.